huaweicloud/services/cdn/data_source_huaweicloud_cdn_domains.go

```diff
@@ -0,0 +1,307 @@
+		getCDNDomainsResp, err := getCDNDomainsClient.Request("GET", currentPath, &getCDNDomainsOpt)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		getCDNDomainsRespBody, err := utils.FlattenResponse(getCDNDomainsResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		domains := utils.PathSearch("domains", getCDNDomainsRespBody, make([]interface{}, 0)).([]interface{})
+		rst = append(rst, flattenListDomainsBody(domains)...)
+		if len(domains) == 0 {
+			break
+		}
+		currentTotal++
+	}
+	dataSourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(dataSourceId)
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("domains", filterGetDomainsResponseBodyFlavor(rst, d)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenListDomainsBody(domains []interface{}) []interface{} {
+	rst := make([]interface{}, 0)
+	for _, v := range domains {
+		createTime := utils.PathSearch("create_time", v, 0)
+		updateTime := utils.PathSearch("modify_time", v, 0)
+		rst = append(rst, map[string]interface{}{
+			"id":                            utils.PathSearch("id", v, nil),
+			"name":                          utils.PathSearch("domain_name", v, nil),
+			"type":                          utils.PathSearch("business_type", v, nil),
+			"domain_status":                 utils.PathSearch("domain_status", v, nil),
+			"cname":                         utils.PathSearch("cname", v, nil),
+			"sources":                       flattenSources(v),
+			"domain_origin_host":            utils.PathSearch("domain_origin_host", v, nil),
+			"https_status":                  utils.PathSearch("https_status", v, nil),
+			"created_at":                    utils.FormatTimeStampRFC3339(int64(createTime.(float64))/1000, false),
+			"updated_at":                    utils.FormatTimeStampRFC3339(int64(updateTime.(float64))/1000, false),
+			"disabled":                      utils.PathSearch("disabled", v, nil),
+			"locked":                        utils.PathSearch("locked", v, nil),
+			"auto_refresh_preheat":          utils.PathSearch("auto_refresh_preheat", v, nil),
+			"service_area":                  utils.PathSearch("service_area", v, nil),
+			"range_based_retrieval_enabled": converseStatusToBool(utils.PathSearch("range_status", v, nil).(string)),
+			"follow_status":                 utils.PathSearch("follow_status", v, nil),
+			"origin_status":                 utils.PathSearch("origin_status", v, nil),
+			"banned_reason":                 utils.PathSearch("banned_reason", v, nil),
+			"locked_reason":                 utils.PathSearch("locked_reason", v, nil),
+			"enterprise_project_id":         utils.PathSearch("enterprise_project_id", v, nil),
+			"tags":                          utils.FlattenTagsToMap(utils.PathSearch("tags", v, make(map[string]interface{}, 0))),
+		})
+	}
+	return rst
+}
+
+func converseStatusToBool(status interface{}) bool {
+	return status == "on" || status == 1
```

Comment:
What does "status == 1" mean?

Reply:
The value of field obs_web_hosting_enabled is int. The value of this field in resource is bool. Datasource needs to be synchronized with resource.